app/models/manageiq/providers/embedded_ansible/crud_common.rb

```diff
@@ -0,0 +1,112 @@
+module ManageIQ::Providers::EmbeddedAnsible::CrudCommon
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def queue(zone, instance_id, method_name, args, action, auth_user)
+      task_opts = {
+        :action => action,
+        :userid => auth_user || "system"
+      }
+
+      queue_opts = {
+        :args        => args,
+        :priority    => MiqQueue::HIGH_PRIORITY,
+        :class_name  => name,
+        :method_name => method_name,
+        :role        => "ems_operations", # TODO: This should go to a git_owner
+        :zone        => zone
+      }
+      queue_opts[:instance_id] = instance_id if instance_id
+      MiqTask.generic_action_with_callback(task_opts, queue_opts)
+    end
+
+    def notify(op_type, manager_id, params)
+      error = nil
+      yield
+    rescue StandardError => e
+      _log.debug(error.result.error) if e.kind_of?(AwesomeSpawn::CommandResultError)
+      raise
+    ensure
+      send_notification(op_type, manager_id, params, error.nil?) if notify_on_provider_interaction?
+    end
+
+    def notify_on_provider_interaction?
+      false
+    end
+
+    def raw_create_in_provider(_manager, _params)
+      raise NotImplementedError, "must be implemented in a subclass"
+    end
+
+    def create_in_provider(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      notify('creation', manager_id, params) do
+        raw_create_in_provider(manager, params)
+      end
+    end
+
+    def create_in_provider_queue(manager_id, params, auth_user = nil)
```

Comment:
Yeah I agree... Given the change in nature of embedded ansible, there are a LOT of places like this that we can clean up in the future.

Reply:
Had a bit of a realization with this one while doing some rebasing:  We might not want to move this out of the queue since something like creating a new `configuration_script_source` would need to do a `git clone`, and depending on the size of the repo, we don't want to be holding up a UI thread doing that.

Possibly we could figure out a way to selectively do that with `crud_common.rb`, but that is a implementation for our future selves.